org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend

```diff
@@ -90,27 +92,19 @@ class CodeMiningFragment extends AbstractStubGeneratingFragment {
 			import org.eclipse.jface.text.IDocument
 			import org.eclipse.jface.text.codemining.ICodeMining
 			import org.eclipse.xtext.resource.XtextResource
-			import org.eclipse.xtext.ui.codemining.XtextCodeMiningProvider
+			import org.eclipse.xtext.ui.codemining.AbstractXtextCodeMiningProvider
 			import org.eclipse.xtext.util.CancelIndicator
 			import org.eclipse.xtext.util.IAcceptor
 			
-			class �codeMiningProviderClass.lastSegment� extends XtextCodeMiningProvider {
-				override void createLineHeaderCodeMinings(IDocument document, XtextResource resource, CancelIndicator indicator, IAcceptor<ICodeMining> acceptor) throws BadLocationException{
-					
-					//TODO: implement me
-					//use acceptor.accept(super.createNewLineHeaderCodeMining(...)) to add a new code mining to the final list
-					
-					//example:
-					//acceptor.accept(createNewLineHeaderCodeMining(1, document, "Header annotation"))
-				}
-				
-				override void createLineContentCodeMinings(IDocument document, XtextResource resource, CancelIndicator indicator, IAcceptor<ICodeMining> acceptor)  throws BadLocationException {
+			class �codeMiningProviderClass.lastSegment� extends AbstractXtextCodeMiningProvider {
+				override void createCodeMinings(IDocument document, XtextResource resource, CancelIndicator indicator,
```

Comment:
@szarnekow  @kthoms isnt it convention to use "typeRefs" or <<XtextResource>> at such places?

Reply:
(instead of using the imports?)

Reply:
True, that would be cleaner.

Reply:
done.